frontend/src/Hooks/useEditWorkshop.ts

```diff
@@ -19,50 +22,95 @@ export default function useEditWorkshop(props: EditWorkshopFormProps){
     const [workshopName, setWorkshopName]
         = useState<string>(props.workshopToEdit?.name ?? (props.user?.username || ""))
+    const [address, setAddress] = useState(props.workshopToEdit?.location ?? "")
+    const [coordinates, setCoordinates]
+        = useState<Coordinates | undefined>(props.workshopToEdit?.coordinates ?? undefined)
     const [addComponentDialogOpen, setAddComponentDialogOpen] = useState(false)
-
+    const [invalidAddress, setInvalidAddress] = useState(false)
     function handleServicesChange(event: SyntheticEvent, value: string[]) {
         setServices(value)
-        console.log(event.target)
     }
     function handleWorkshopNameChange(event: ChangeEvent<HTMLInputElement>) {
         setWorkshopName(event.target.value)
     }
     function handleSetComponents(components: Component[]){
         setComponents(components)
     }
+    function handleAddressChange(event: ChangeEvent<HTMLInputElement>){
+        setAddress(event.target.value)
+        setInvalidAddress(false)
+    }
     function handleSetOpenAddComponentsDialog(){
         setAddComponentDialogOpen(!addComponentDialogOpen)
     }
-    function handleSubmit(event: FormEvent<HTMLFormElement>){
+    function getCoordinates(){
+        return axios.get(`https://nominatim.openstreetmap.org/search?format=jsonv2&limit=1&q=${address}`)
+            .then(r => r.data as OSMSearchResult[])
+            .then(results => {
+                if(results.length>0){
+                    setCoordinates({lat: Number(results[0].lat), lng: Number(results[0].lon)})
+                } else {
+                    toast.error("Invalid Address")
+                    setInvalidAddress(true)
+                }
+                return results
+            })
+    }
+
+    function handleSubmit(event: FormEvent<HTMLFormElement>) {
         event.preventDefault()
-        if(!props.workshopToEdit) {
-            axios.post("/api/workshops/",
-                {id: props.user?.id, name: workshopName, services: services, inventory: components})
-                .then(r=> r.data)
-                .then((newWorkshop)=>props.updateWorkshopList([...props.workshops, newWorkshop]))
-                .then(()=> navigate("/"))
-                .catch((error) => console.error(error))
-        } else {
-            axios.put("/api/workshops/"+ props.workshopToEdit.id,
-                {id: props.user?.id, name: workshopName, services: services, inventory: components})
-                .then(r=> r.data)
-                .then(updatedWorkshop=> props.updateWorkshopList(
-                    [...props.workshops.filter(workshop => workshop.id!==updatedWorkshop.id),
-                        updatedWorkshop]))
-                .then(()=> navigate("/"))
-                .catch((error) => console.error(error))
-        }
+        getCoordinates()
+            .then((results)=> {
+                if(results.length<1){
+                    return results
```

Comment:
Why return the results of length 0? I think a simple `return` would be clearer (or I just misunderstood the code, given the time of the day)